gulpfile.js

```diff
@@ -118,6 +128,34 @@ gulp.task('build-ci', ['clean'], function (cb) {
     runSequence('build', cb);
 });
 
+gulp.task('build-test', function buildTest(cb) {
+    var config = _.defaultsDeep({}, require('./webpack.config'));
+
+    config.entry = {};
+    // the key in the entry, formed by path.join, determines the target folder structure of the test file
+    config.entry[path.join('request', 'repository', 'request-repository-cache')] = './test/request/repository/request-repository-cache.spec.ts';
+
+    config.output.path = testSettings.output;
+
+    var started = false;
+    function processResult(err, stats) {
+        gutil.log('Webpack\n' + stats.toString(config.log));
+
+        if (!started) {
+            started = true;
+            cb();
+        }
+    }
+
+    var compiler = webpack(config);
+    compiler.run(processResult);
+});
```

Comment:
This runs the whole `webpack` build... I know we talked about the need to inject fake/mock modules. Are we looking at dealing with that yet or are we going to bring that online later. Note, I know we change the `config.entry` point, but how are we mocking these. 

Additionally, at the moment it looks like we need to manually target specific `*.spec`s, is this just temporary or are we looking to change this later on.